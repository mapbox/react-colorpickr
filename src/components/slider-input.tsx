import React, { CSSProperties } from 'react';
import * as SliderPrimitive from '@radix-ui/react-slider';

interface Props {
  onChange: (n: number) => void;
  colorValue: string;
  value: number;
  min: number;
  max: number;
  id: string;
  trackStyle: CSSProperties;
  disabled: boolean;
}

function SliderInput({
  onChange,
  disabled,
  colorValue,
  value,
  min,
  max,
  id,
  trackStyle
}: Props) {
  const rootProps = {
    id,
    name: id,
    onValueChange: (value: Array<number>) => onChange(value[0]),
    value: [value],
    min,
    max,
    disabled,
    className: 'relative flex h12 w-full round-full',
    style: {
      cursor: disabled ? 'default' : 'pointer',
      alignItems: 'center',
      touchAction: 'none'
    },
    'data-testid': `${id}-slider`
  };

  if (disabled) {
    rootProps.className = `${rootProps.className} opacity50`;
  }

  return (
    <SliderPrimitive.Root {...rootProps}>
      <div className="absolute w-full h3 bg-tile" />
      <SliderPrimitive.Track
        style={{ ...trackStyle }}
        className="h3 relative flex-child-grow round-full"
      />
      <SliderPrimitive.Thumb
        className="w12 h12 round-full block border border--white shadow-darken10"
        style={{
          cursor: disabled ? 'default' : 'ew-resize',
          backgroundColor: colorValue
        }}
      />
    </SliderPrimitive.Root>
  );
}

export { SliderInput };
