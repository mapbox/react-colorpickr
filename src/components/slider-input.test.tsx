import React from 'react';
import { render, screen } from '@testing-library/react';
import { SliderInput } from './slider-input';

describe('SliderInput', () => {
  const props = {
    onChange: jest.fn(),
    colorValue: 'red',
    trackStyle: { backgroundColor: 'green' },
    id: 'hue',
    value: 50,
    min: 0,
    max: 360,
    disabled: false
  };

  test('renders', () => {
    const { baseElement } = render(<SliderInput {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  test('disabled', () => {
    render(<SliderInput {...props} disabled={true} />);
    expect(
      screen.getByTestId('hue-slider').getAttribute('data-disabled')
    ).not.toEqual(null);
  });
});
