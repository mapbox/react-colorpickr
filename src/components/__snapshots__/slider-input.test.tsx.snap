// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderInput renders 1`] = `
<body>
  <div>
    <span
      aria-disabled="false"
      class="relative flex h12 w-full round-full cursor-pointer"
      data-orientation="horizontal"
      data-testid="hue-slider"
      dir="ltr"
      id="hue"
      style="align-items: center; --radix-slider-thumb-transform: translateX(-50%);"
    >
      <div
        class="absolute w-full h3 bg-tile"
      />
      <span
        class="h3 relative flex-child-grow round-full"
        data-orientation="horizontal"
        style="background-color: green;"
      />
      <span
        style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(13.88888888888889% + 0px);"
      >
        <span
          aria-orientation="horizontal"
          aria-valuemax="360"
          aria-valuemin="0"
          aria-valuenow="50"
          class="w12 h12 round-full block border border--white shadow-darken10"
          data-orientation="horizontal"
          data-radix-collection-item=""
          role="slider"
          style="cursor: ew-resize; background-color: red;"
          tabindex="0"
        />
      </span>
    </span>
  </div>
</body>
`;
